apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl

```diff
@@ -620,8 +645,9 @@ producers_config(BridgeType, BridgeName, ClientId, Input, IsDryRun, BridgeV2Id)
 partitioner(random) -> random;
 partitioner(key_dispatch) -> first_key_dispatch.
 
-replayq_dir(ClientId) ->
-    filename:join([emqx:data_dir(), "kafka", ClientId]).
+replayq_dir(BridgeType, BridgeName) ->
+    DirName = iolist_to_binary([BridgeType, ":", BridgeName, ":", atom_to_list(node())]),
```

Comment:
Note: in v2, kafka producer is now called `kafka_producer`, but in the old impl it was just `kafka`.  So we may need to downgrade the bridge type here to make it backwards compatible.